napari/layers/base/base.py

```diff
@@ -983,34 +984,39 @@ def _data_slice(self) -> _ThickNDSlice:
     def _get_ndim(self) -> int:
         raise NotImplementedError
 
-    def _get_base_state(self) -> dict:
+    def _get_base_state(self) -> DeprecatingDict:
         """Get dictionary of attributes on base layer.
 
+        This is useful for serialization and deserialization of the layer.
+        And similarly for plugins to pass state without direct dependencies on napari types.
+
         Returns
         -------
-        state : dict
-            Dictionary of attributes on base layer.
+        DeprecatingDict
+            Dictionary of attributes on base layer that issues warning messages when
+            deprecated keys are accessed directly.
         """
-        base_dict = {
-            'name': self.name,
-            'metadata': self.metadata,
-            'scale': list(self.scale),
-            'translate': list(self.translate),
-            'rotate': [list(r) for r in self.rotate],
-            'shear': list(self.shear),
-            'affine': self.affine.affine_matrix,
-            'opacity': self.opacity,
-            'blending': self.blending,
-            'visible': self.visible,
-            'experimental_clipping_planes': [
-                plane.dict() for plane in self.experimental_clipping_planes
-            ],
-            'projection_mode': self.projection_mode,
-        }
-        return base_dict
+        return DeprecatingDict(
```

Comment:
I personally prefer to create here dict, and convert it to `DeprecatingDict` in `Points._get_state`.

Reply:
Good idea. Especially since `Points` is the only class that uses this.